tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+            if default_value is not None:
+                cli = option_struct.cli_subcommand()
+                _, config = cli.parse()
+                option_value = config.get_config().get(
+                    long_option_with_underscores).default_value()
+                self.assertEqual(option_value, default_value)
+
+            # Verify that a incorrect value causes a failure
+            if expected_failing_value is not None:
+                cli = option_struct.cli_subcommand()
+                cli.args.extend([long_option, expected_failing_value])
+                with self.assertRaises(SystemExit):
+                    _, config = cli.parse()
+
+    def _test_list_option(self, option_struct):
+        long_option = option_struct.long_flag
+        short_option = option_struct.short_flag
+        expected_value = option_struct.expected_value
+        default_value = option_struct.default_value
+
+        # Convert expected and default values to proper types for comparison
+        if option_struct.type == "intlist":
+            expected_value_converted = self._convert_string_to_int_list(
+                expected_value)
+            if default_value is not None:
+                default_value_converted = self._convert_string_to_int_list(
+                    default_value)
+        else:
+            expected_value_converted = expected_value.split(",")
+            expected_value_converted = self._convert_string_to_string_list(
+                expected_value)
+            if default_value is not None:
+                default_value_converted = self._convert_string_to_string_list(
+                    default_value)
+
+        long_option_with_underscores = self._convert_flag(long_option)
+
+        # Test the long flag
+        cli = option_struct.cli_subcommand()
+        cli.args.extend([long_option, expected_value])
+        if option_struct.extra_commands is not None:
+            cli.args.extend(option_struct.extra_commands)
+        _, config = cli.parse()
+        option_value = config.get_config().get(
+            long_option_with_underscores).value()
+        self.assertEqual(option_value, expected_value_converted)
+
+        # Test the short flag
+        if short_option is not None:
+            cli = option_struct.cli_subcommand()
+            cli.args.extend([short_option, expected_value])
+            _, config = cli.parse()
+            option_value = config.get_config().get(
+                long_option_with_underscores).value()
+            self.assertEqual(option_value, expected_value_converted)
+
+        # Verify the default value for the option
+        if default_value is not None:
+            cli = option_struct.cli_subcommand()
+            _, config = cli.parse()
+            option_value = config.get_config().get(
+                long_option_with_underscores).default_value()
+            self.assertEqual(option_value, default_value_converted)
+
+    # Helper methods
+    def _convert_flag(self, option):
```

Comment:
convert_flag_to_?

Reply:
the internal representation that the config fields use.